Code/Framework/AzFramework/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad.cpp

```diff
@@ -144,8 +151,11 @@ namespace AzFramework
             m_thumbStickDirectionChannelsById[channelId] = channel;
         }
 
-        // Create the platform specific implementation
-        m_pimpl.reset(Implementation::Create(*this));
+        // Create a custom implementation if we've been provided one,
+        // otherwise default to the platform specific implementation
```

Comment:
question: It is possible to do something like this?

```
// .h
InputDeviceGamepad(
    const InputDeviceId& inputDeviceId,
    ImplementationFactory implementationFactory = &Implementation::Create)

// .cpp
m_pimpl.reset(implementationFactory(*this));
```

So we don't have to have the `nullptr` checks each time?

This might not be possible because of needing the implementation of the `Implementation` type, but just wanted to see as it might simplify the `reset` call slightly

Reply:
Great suggestion, updated. I like that doing it this way allows for a nullptr implementation to be specified by simply passing in a nullptr implementationFactory, rather than a pointer to a function that returns nullptr:)

Reply:
Awesome! 😄 That sounds great thanks!